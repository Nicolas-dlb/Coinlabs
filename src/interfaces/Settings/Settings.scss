.settings {
  padding-left: 10px;
  padding-bottom: 20px;
  border-radius: 30px;
  background: var(--secondary-bg-color);
  padding-top: 10vh;
  opacity: 0;
  pointer-events: none;
  margin-top: 1vh;
  overflow: hidden;
  position: relative;
  bottom: 489vh;
  .page-title {
    margin-left: 10px;
  }
  .btn_settings {
    background: #2b263f;
    outline: none;
    border: none;
    border-radius: 5px;
    padding: 7px 15px;
    height: fit-content;
    color: var(--primary-font-color);
    font-family: inherit;
    font-size: 15px;
    cursor: pointer;
  }

  .input_settings {
    display: flex;
    margin-top: 2vh;
    border: 1px solid var(--third-font-color);
    border-radius: 10px;

    width: 60%;
    align-items: center;
    p {
      margin-left: 10px;
      font-size: 19px;
      color: var(--primary-font-color);
    }
    input {
      background: var(--third-bg-color);
      border: none;
      outline: none;
      color: var(--primary-font-color);
      caret-color: #1baa63;
      border-radius: 10px;
      font-family: inherit;
      padding: 8px;

      &::placeholder {
        font-family: inherit;
        color: #484558;
        font-size: 12px;
      }
    }
  }
  .settings_top_right {
    display: none;
  }
}

@media only screen and (min-width: 992px) {
  .settings {
    padding-left: 208px;
    bottom: 441vh;
    scale: 0.97;
    .settings_theme:last-child {
      display: none;
    }
    .page-title {
      margin-left: 1%;
    }

    .settings_top {
      display: flex;
      .settings_top_right {
        display: flex;
        width: 64.5%;
        margin-left: 10px;
        .settings_theme {
          display: flex;
        }
        .theme {
          width: 100%;
        }
      }
    }
  }
}
