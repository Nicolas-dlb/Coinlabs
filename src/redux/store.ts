import { configureStore } from "@reduxjs/toolkit";
import marketReducer from "./reducers/marketSlice";
import walletsReducer from "./reducers/walletsSlice";
import appReducer from "./reducers/appSlice";
import userReducer from "./reducers/userSlice";

export default configureStore({
  reducer: {
    market: marketReducer,
    wallets: walletsReducer,
    app: appReducer,
    user: userReducer,
  },
});
