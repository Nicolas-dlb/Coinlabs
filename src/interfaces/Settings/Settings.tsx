/* eslint-disable no-unused-vars */
import { auth, db } from "firebaseConfig";
import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import {
  selectUserEmail,
  selectUserName,
  setProfilPic,
  setUserEmail,
  setUsername,
} from "redux/reducers/userSlice";
import "./Settings.scss";
import $ from "jquery";
import { selectProfilPicture } from "utils/utils";
import { selectWallets, setWallets } from "redux/reducers/walletsSlice";
import * as admin from "firebase-admin";
import Account from "./Account/Account";
import Picture from "./Picture/Picture";
import Username from "./Username/Username";
import Email from "./Email/Email";
import Founds from "./Founds/Founds";
import Theme from "./Theme/Theme";

function Settings() {
  const dispatch = useDispatch();
  const username = useSelector(selectUserName);

  const userEmail = useSelector(selectUserEmail);

  const [userName, setUserName] = useState(username);

  const [picture, setPicture] = useState("");

  useEffect(() => {
    if (username) {
      setUserName(username);
      $("#change_username").val(username);
    }
  }, [username]);

  const changeUsername = () => {
    const value: any = $("#change_username").val();

    db.collection("users").doc(auth.currentUser?.uid).set(
      {
        username: value,
      },
      { merge: true }
    );
    dispatch(setUsername(value));
    setUserName(value);
    $("#change_username").val();
  };

  useEffect(() => {
    dispatch(setProfilPic(picture));
    db.collection("users").doc(auth.currentUser?.uid).set(
      {
        profilPic: picture,
      },
      { merge: true }
    );
  }, [picture]);

  return (
    <div id="settings" className="settings">
      <p className="page-title">Settings</p>
      <div className="settings_top">
        <div className="settings_top_left">
          <Account />
        </div>
        <div className="settings_top_right">
          <Theme />
        </div>
      </div>
      <Picture setPicture={setPicture} />
      <Username onClick={changeUsername} />
      <Email />
      <Founds />
      <Theme />
    </div>
  );
}

export default Settings;
