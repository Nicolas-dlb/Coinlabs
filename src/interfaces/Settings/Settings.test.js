import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import store from "redux/store";
import { Provider } from "react-redux";
import Settings from "./Settings";

describe("settings", () => {
  let SettingsComponent;
  beforeEach(async () => {
    await act(async () => {
      SettingsComponent = await mount(
        <Provider store={store}>
          <Settings />
        </Provider>
      );
    });
  });
  it("should render correctly", () => {
    expect(SettingsComponent).toMatchSnapshot();
  });

  it("should render correctly account panel", () => {
    expect(SettingsComponent.find(".settings_account").exists()).toBe(true);
  });
  it("should render correctly username panel", () => {
    expect(SettingsComponent.find(".settings_username").exists()).toBe(true);
  });
  it("should render correctly picture panel", () => {
    expect(SettingsComponent.find(".settings_picture").exists()).toBe(true);
  });
  it("should render correctly email panel", () => {
    expect(SettingsComponent.find(".settings_email").exists()).toBe(true);
  });
  it("should render correctly founds panel", () => {
    expect(SettingsComponent.find(".settings_founds").exists()).toBe(true);
  });
  it("should render correctly theme panel", () => {
    expect(SettingsComponent.find(".settings_theme").exists()).toBe(true);
  });
});
