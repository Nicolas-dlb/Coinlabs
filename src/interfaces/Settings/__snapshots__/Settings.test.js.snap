// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings should render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Settings>
    <div
      className="settings"
      id="settings"
    >
      <p
        className="page-title"
      >
        Settings
      </p>
      <div
        className="settings_top"
      >
        <div
          className="settings_top_left"
        >
          <Account>
            <div
              className="settings_account"
            >
              <h3
                className="account"
              >
                Account
              </h3>
              <div
                className="user_card"
              >
                <div
                  className="user_card_img"
                >
                  <img
                    alt=""
                    src="https://www.vhv.rs/dpng/d/164-1645859_selfie-clipart-groucho-glass-good-profile-hd-png.png"
                  />
                  <div
                    className="point"
                  />
                </div>
                <div
                  className="user_card_info"
                >
                  <h3 />
                  <h5 />
                  <div
                    className="change_user"
                  >
                    <p>
                      Not you?
                    </p>
                    <span
                      className="change"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      Change User
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </Account>
        </div>
        <div
          className="settings_top_right"
        >
          <Theme>
            <div
              className="settings_theme"
            >
              <h3>
                Theme
              </h3>
              <div
                className="settings_theme_container"
              >
                <button
                  className="btn-theme btn_theme_active"
                  onClick={[Function]}
                  type="button"
                >
                  Dark
                </button>
                <button
                  className="btn-theme"
                  onClick={[Function]}
                  type="button"
                >
                  Grey
                </button>
              </div>
            </div>
          </Theme>
        </div>
      </div>
      <Picture
        setPicture={[Function]}
      >
        <div
          className="settings_picture"
        >
          <h3>
            Profil picture
          </h3>
          <div
            className="settings_picture_container"
          >
            <div
              className="settings_picture_input"
            >
              <button
                className="set_profil_picture"
                onClick={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
                type="button"
              >
                Select file
              </button>
              <input
                accept="image/jpeg"
                id="picture-input"
                onChange={[Function]}
                style={
                  Object {
                    "display": "none",
                  }
                }
                type="file"
              />
            </div>
            <div
              className="settings_picture_message"
              id="set_picture_message"
            >
              <p>
                Any file selected
              </p>
            </div>
          </div>
        </div>
      </Picture>
      <Username
        onClick={[Function]}
      >
        <div
          className="settings_username"
        >
          <h3>
            Username
          </h3>
          <div
            className="settings_username_input"
          >
            <div
              className="input"
            >
              <p>
                @
              </p>
              <input
                autoComplete="off"
                className="username_element"
                id="change_username"
                placeholder="Enter username"
                spellCheck="false"
                type="text"
              />
            </div>
            <button
              className="btn_settings"
              onClick={[Function]}
              type="button"
            >
              Update
            </button>
          </div>
        </div>
      </Username>
      <Email
        setEmail={[Function]}
      >
        <div
          className="settings_email"
        >
          <h3>
            Email
          </h3>
          <div
            className="settings_email_input"
          >
            <div
              className="input input_settings"
            >
              <svg
                className="email_icon"
                height="489"
                viewBox="0 0 489.2 489.2"
                width="489"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="0 386.6 143.5 214.9 0 99.6 "
                />
                <polygon
                  points="489.2 386.6 489.2 99.6 345.6 214.9 "
                />
                <polygon
                  points="487.9 398.7 489.2 398.7 489.2 397.7 "
                />
                <polygon
                  points="0 398.7 1.2 398.7 0 397.7 "
                />
                <polygon
                  points="480.8 398.7 479.9 397.7 334.6 223.7 244.6 296.1 154.6 223.7 9.2 397.7 8.4 398.7 "
                />
                <polygon
                  points="152.6 204 158.1 208.5 163.6 212.9 244.6 277.9 325.5 212.9 331 208.5 336.5 204 477.9 90.5 11.3 90.5 "
                />
              </svg>
              <input
                id="set_email"
                onChange={[Function]}
                placeholder="Enter new email"
                type="text"
              />
            </div>
            <button
              className="btn_settings"
              type="button"
            >
              Update
            </button>
          </div>
        </div>
      </Email>
      <Founds>
        <div
          className="settings_founds"
        >
          <h3>
            Founds
          </h3>
          <div
            className="settings_founds_input"
          >
            <div
              className="input"
            >
              <p>
                $
              </p>
              <input
                id="add_founds"
                min="0"
                onChange={[Function]}
                placeholder="Enter amount"
                type="number"
              />
            </div>
            <button
              className="btn_settings"
              onClick={[Function]}
              type="button"
            >
              Add
            </button>
          </div>
        </div>
      </Founds>
      <Theme>
        <div
          className="settings_theme"
        >
          <h3>
            Theme
          </h3>
          <div
            className="settings_theme_container"
          >
            <button
              className="btn-theme btn_theme_active"
              onClick={[Function]}
              type="button"
            >
              Dark
            </button>
            <button
              className="btn-theme"
              onClick={[Function]}
              type="button"
            >
              Grey
            </button>
          </div>
        </div>
      </Theme>
    </div>
  </Settings>
</Provider>
`;
